server/webapp/WEB-INF/rails.new/app/controllers/api_v2/admin/templates_controller.rb

```diff
@@ -42,7 +42,7 @@ def create
 
       def update
         result = HttpLocalizedOperationResult.new
-        updated_template = ApiV2::Config::TemplateConfigRepresenter.new(PipelineTemplateConfig.new).from_hash(params[:template])
+        updated_template = ApiV2::Config::TemplateConfigRepresenter.new(@template).from_hash(params[:template])
```

Comment:
This might actually be a very dangerous thing to do... If the update fails midway, you'd have still applied partial changes onto the actual object as part of the deserialization call.